packages/core/src/__tests__/unit/application-lifecycle.unit.ts

```diff
@@ -0,0 +1,148 @@
+import {expect} from '@loopback/testlab';
+import {
+  Application,
+  Component,
+  CoreBindings,
+  CoreTags,
+  LifeCycleObserver,
+  Server,
+} from '../..';
+
+describe('Application life cycle', () => {
+  describe('start', () => {
+    it('starts all injected servers', async () => {
+      const app = new Application();
+      app.component(FakeComponent);
+      const component = await app.get<FakeComponent>(
+        `${CoreBindings.COMPONENTS}.FakeComponent`,
+      );
+      expect(component.status).to.equal('not-initialized');
+      await app.start();
+      const server = await app.getServer(FakeServer);
+
+      expect(server).to.not.be.null();
+      expect(server.listening).to.equal(true);
+      expect(component.status).to.equal('started');
+      await app.stop();
+    });
+
+    it('starts servers bound with `LIFE_CYCLE_OBSERVER` tag', async () => {
+      const app = new Application();
+      app
+        .bind('fake-server')
+        .toClass(FakeServer)
+        .tag(CoreTags.LIFE_CYCLE_OBSERVER, CoreTags.SERVER)
+        .inScope(BindingScope.SINGLETON);
+      await app.start();
+      const server = await app.get<FakeServer>('fake-server');
+
+      expect(server).to.not.be.null();
+      expect(server.listening).to.equal(true);
+      await app.stop();
+    });
+
+    it('starts/stops all registered components', async () => {
+      const app = new Application();
+      app.component(FakeComponent);
+      const component = await app.get<FakeComponent>(
+        `${CoreBindings.COMPONENTS}.FakeComponent`,
+      );
+      expect(component.status).to.equal('not-initialized');
+      await app.start();
+      expect(component.status).to.equal('started');
+      await app.stop();
+      expect(component.status).to.equal('stopped');
+    });
+
+    it('starts/stops all observers from the component', async () => {
+      const app = new Application();
+      app.component(FakeComponentWithAnObserver);
+      const observer = await app.get<MyObserver>(
+        'lifeCycleObservers.MyObserver',
+      );
+      expect(observer.status).to.equal('not-initialized');
+      await app.start();
+      expect(observer.status).to.equal('started');
+      await app.stop();
+      expect(observer.status).to.equal('stopped');
+    });
+
+    it('starts/stops all registered life cycle observers', async () => {
+      const app = new Application();
+      app.lifeCycleObserver(MyObserver, 'my-observer');
+
+      const observer = await app.get<MyObserver>(
+        'lifeCycleObservers.my-observer',
+      );
+      expect(observer.status).to.equal('not-initialized');
+      await app.start();
+      expect(observer.status).to.equal('started');
+      await app.stop();
+      expect(observer.status).to.equal('stopped');
+    });
+
+    it('does not attempt to start poorly named bindings', async () => {
+      const app = new Application();
+      app.component(FakeComponent);
```

Comment:
I don't see how is FakeComponent relevant to this particular test case. Can you remove this line please?